src/boot/is_debug.c

```diff
@@ -48,7 +48,7 @@ void* is_proutSyncPrintf(void* arg, const char* str, u32 count) {
 
     osEPiReadIo(sISVHandle, (u32)&gISVDbgPrnAdrs->magic, &data);
     if (data != ASCII_TO_U32('I', 'S', '6', '4')) {
-        return 1;
+        return (void*)1;
```

Comment:
The function is used as a `PrintCallback`, which is typed with a `void*` return. However this function doesn't use `arg` so all that matters is whether it returns NULL or not, which is why `(void*)1` works fine afaik (see `_Printf` and more specifically `_PROUT` for the callback usage)

Reply:
Yeah I mean changing all types (arg and the return) to s32, matches

Reply:
Hmm that wouldn't fit the real type afaik, and it would require integer casts wherever the argument and return type are used as pointers (eg. `GfxPrint_Callback`), so it seems worse overall

Reply:
It's not correct for it to be anything but `void*`, the intention is to be able to pass any kind of user data through the 2nd arg of `_Printf`.
Consider `proutSprintf` in `sprintf.c`, it uses `arg` as a pointer to memcpy from and returns the next pointer to memcpy from. I also wrote an `snprintf` implementation that uses `arg` and the return to pass a pointer to a struct around: https://pastebin.com/qzShRtpf
Keeping it as a pointer would be the most generally useful.

Reply:
I didn't see it was actually used elsewhere oops

The GfxPrint usage looks super jank, `PrintUtils_VPrintf` passes `PrintCallback* pfn` as `arg`, and `GfxPrint_Callback` later casts `arg` back to `GfxPrint* this`, which only works because `offsetof(GfxPrint, callback) == 0`